include/pisa/topk_queue.hpp

```diff
@@ -34,18 +51,21 @@ struct topk_queue {
         if (PISA_UNLIKELY(m_q.size() <= m_k)) {
             std::push_heap(m_q.begin(), m_q.end(), min_heap_order);
             if (PISA_UNLIKELY(m_q.size() == m_k)) {
-                m_threshold = m_q.front().first;
+                m_effective_threshold = m_q.front().first;
```

Comment:
Wouldn't this create slowdowns where a good initial threshold is provided?

As in, assume the `initial_threshold = 10` and then once the heap has *k* elements the *k* th score is `5.0` - this is probably not what is desired.

Reply:
I'm not sure I understand. Do you mean that the threshold would decrease? I don't think it can, because only higher scores would go past `would_enter` at the top of the function, right? Or am I missing something?

Reply:
Ah, yes, I think you are right; we can only get a change in `m_effective_threshold` once k elements enter, but those elements can only enter if they exceed that value in the first place. My mistake!

But hey, at least we have a test for that now (thanks!)